src/monotonic/monotonic.jl

```diff
@@ -279,6 +296,33 @@ function calcTangents(::Type{TCoeffs}, x::AbstractVector{<:Number},
     return (m, Δ)
 end
 
+function calcTangents(::Type{TCoeffs}, x::AbstractVector{<:Number},
+    y :: AbstractVector{TEl}, method::AkimaMonotonicInterpolation) where {TCoeffs, TEl}
+
+    # based on Akima (1970),
+    # "A New Method of Interpolation and Smooth Curve Fitting Based on Local Procedures",
+    # doi:10.1145/321607.321609.
+
```

Comment:
Check if `x` and `y` are the same the length. Throw an `ArgumentError` otherwise. Maybe this is already done elsewhere?

Reply:
You are right here, the check is performed somewhere else:
The not exported `calcTangents` method is called while calling the exported `interpolate` method, which in turn calls `check_monotonic`. Hence when we call `interpolate( [1,2,3,4,5], [1,2,3,4,5,6], AkimaMonotonicInterpolation())` we will get the error

> ERROR: DimensionMismatch: knot vector must have the same axes as the corresponding array

Therefore everything is already fine.